src/layer.rs

```diff
@@ -30,6 +30,16 @@ pub trait Layer: Send + Debug + erased_serde::Serialize {
         bytes: &[u8],
     ) -> Result<(Option<Box<dyn Layer + Send>>, usize), Error>;
 
+    /// Main 'encoder' function.
+    ///
+    /// The return value is a `Vec<u8>` on success. This indicates the encoded packet of the
+    /// layer. Typically return a [SculptingError][`crate::errors::Error::SculptingError`],
+    /// but may as well return other values. This updates the fields of the packet using the
+    /// encoded bytes of the next layer(`next_layer`), and `info`, to provide additional details
+    /// about the packet. This can be the result of [Layer::name()][`Self::name()`]
+    /// of the next packet.
+    fn stack_and_encode(&mut self, next_layer: &[u8], info: &str) -> Result<Vec<u8>, Error>;
```

Comment:
Sorry I missed this before just create a default `unimplemented!()` implementation here so that you don't have to do it for every `Layer`. Let's do that so that we can just keep the code specific to modified layers, viz. `Ethernet` and `Ipv4` here. 

So this should look like - 

```rust
fn stack_and_encode(&mut self, next_layer: &[u8], info: &str) -> Result<Vec<u8>, Error> {
    unimplemented!()
}